src/select/select.tsx

```diff
@@ -1,142 +1,121 @@
+import * as React from 'react';
+
+import { matches } from '@ampel-ui/common/search';
+
 import { isEqual } from 'lodash';
 
-import * as React from 'react';
+import { onOuterClick } from '@ampel-ui/subscriptions';
 
 import { Option } from '../api/index';
-import { matches } from '../common/search';
-import { onOuterClick } from '../subscriptions';
 
-interface Props<T> {
+import { RendererProps, SelectList } from './select-list';
+
+interface Props<T, O extends Option<T>> {
     id: string;
     value?: T;
-    options: Array<Option<T>>;
+    options: Array<O>;
     onChange: (value: T) => void;
-    disabled?: boolean;
     className?: string;
     searchable?: boolean;
     placeholder?: string;
+    disabled?: boolean;
     disableOptionWhen?: (value: T) => boolean;
-    searchPlaceholder?: string;
+    optionsRenderer?: (props: RendererProps<T, O>) => React.ReactNode;
```

Comment:
Do we have the naming `Renderer` elsewhere? I might be wrong but `renderOptions` seems to fit the current naming better

Reply:
not sure, but we can change the name ofc